### ISSUE TITLE
Blanks are not counted in diff mode

### ISSUE THREAD

**Issue by aakasaka** [AlDanial/cloc#349]
I made two files -  `file1.cs` and `file2.cs` 

file1.cs:
```cs
//comment

Console.WriteLine("Code");
```

file2.cs:
```cs
//comment

Console.WriteLine("Code");
//comment

Console.WriteLine("Code");
```

Then `>cloc file2.cs` outputs below.

```
$ cloc file2.cs
       1 text file.
       1 unique file.
       0 files ignored.

github.com/AlDanial/cloc v 1.78  T=0.07 s (13.4 files/s, 80.6 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
C#                               1              2              2              2
-------------------------------------------------------------------------------
```

I think it's okay, but when I get diff by `>cloc --diff file1.cs file2.cs` , it outputs below.

```
$ cloc --diff file1.cs file2.cs
       1 text file.
       1 text file.
       0 files ignored.

github.com/AlDanial/cloc v 1.78  T=0.07 s (14.7 files/s, 14.7 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
C#
 same                            0              0              1              1
 modified                        1              0              0              0
 added                           0              1              1              1
 removed                         0              0              0              0
-------------------------------------------------------------------------------
SUM:
 same                            0              0              1              1
 modified                        1              0              0              0
 added                           0              1              1              1
 removed                         0              0              0              0
-------------------------------------------------------------------------------
```

I think blanks same should be 1, but it shows 0. Is that by design?

**Comment by AlDanial** [comment]
Your diff output shows the correct result of one blank line added, ref the value ``1`` in the ``blank`` column and ``added`` row:
<pre>
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
C#
 same                            0              0              1              1
 modified                        1              0              0              0
 added                           0              <b>1</b>              1              1
 removed                         0              0              0              0
</pre>

**Comment by aakasaka** [comment]
Not the `added` row, I mean `same` row should be 1, isn't it?

**Comment by AlDanial** [comment]
My mistake, you're right.

**Comment by AlDanial** [comment]
I spent some time digging into this.  I found and fixed a tangentially related issue but it doesn't alter the counts we currently see.
The complication is this:  cloc first strips blank lines before diff'ing source and comments.  As a result it doesn't know the locations of the blank lines, it only has the raw counts of how many such lines are in a file.  Therefore there isn't enough information to determine how many were the same; at most we can determine is the delta--and that's what cloc shows now.
Even the knowledge of the raw counts in each file alone doesn't help.  ``file1.cs`` has one blank line and ``file2.cs`` has two, but cloc has no way of knowing that the first blank line is the "same" in both files.  All it knows is before there was 1 blank and now there are 2 blanks.
I'll update the documentation to point out this limitation.
